Mathlib/SetTheory/Ordinal/Topology.lean

```diff
@@ -227,4 +227,41 @@ theorem enumOrd_isNormal_iff_isClosed (hs : s.Unbounded (· < ·)) :
     rw [hb]
     exact le_bsup.{u, u} _ _ (ha.2 _ hba)
 
+open Set Filter
+
+/-- An ordinal is an accumulation point of a set of ordinals if it is positive and there
+are elements in the set arbitrarily close to the ordinal from below. -/
+def IsAcc (o : Ordinal) (S : Set Ordinal) : Prop :=
+  o ≠ 0 ∧ ∀ p < o, (S ∩ Ioo p o).Nonempty
+
+/-- A set of ordinals is closed below an ordinal if it contains all of
+its accumulation points below the ordinal. -/
+def IsClosed (S : Set Ordinal) (o : Ordinal) : Prop :=
+  ∀ p < o, IsAcc p S → p ∈ S
+
+theorem IsAcc.subset {o : Ordinal} {S T : Set Ordinal} (h : S ⊆ T) (ho : o.IsAcc S) :
+    o.IsAcc T := ⟨ho.1, fun p plto ↦ (ho.2 p plto).casesOn fun s hs ↦ ⟨s, h hs.1, hs.2⟩⟩
+
+theorem IsAcc.isLimit {o : Ordinal} {U : Set Ordinal} (h : o.IsAcc U) : IsLimit o := by
+  refine isLimit_of_not_succ_of_ne_zero (fun ⟨x, hx⟩ ↦ ?_) h.1
+  rcases h.2 x (lt_of_lt_of_le (lt_succ x) hx.symm.le) with ⟨p, hp⟩
+  exact (hx.symm ▸ (succ_le_iff.mpr hp.2.1)).not_lt hp.2.2
+
+theorem IsAcc.inter_Ioo_nonempty {o : Ordinal} {U : Set Ordinal} (hU : o.IsAcc U)
+    {p : Ordinal} (hp : p < o) : (U ∩ Ioo p o).Nonempty := (hU.2 p hp).casesOn fun q hq ↦
+  ⟨q, hq.1, hq.2.1, hq.2.2⟩
+
+theorem isClosed_zero (S : Set Ordinal) : IsClosed S 0 := fun _ h ↦
+  False.elim <| (Ordinal.zero_le _).not_lt h
+
+theorem IsClosed.sInter {o : Ordinal} {S : Set (Set Ordinal)} (h : ∀ C ∈ S, IsClosed C o) :
+    IsClosed (⋂₀ S) o :=
+  fun p plto pAcc C CmemS ↦ (h C CmemS) p plto (pAcc.subset (sInter_subset_of_mem CmemS))
+
+theorem IsClosed.iInter {ι : Type u} {f : ι → Set Ordinal} {o : Ordinal}
+    (h : ∀ i, IsClosed (f i) o) : IsClosed (⋂ i, f i) o := by
+  have := IsClosed.sInter (fun C (⟨i, fieqC⟩ : C ∈ range f) ↦ fieqC ▸ (h i))
```

Comment:
Not on my laptop rn, but can you golf this by doing it backwards? That is, do `apply Isclosed.sInter` and finish by `mem_range` in the next or even the same line.